recipes-support/provisioning/files/test-provision-tpm.sh

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env bash
+#===============================================================================
+#
+#  provision-tpm.sh
+#
+#  Copyright (C) 2021 by PHYTEC Messtechnik GmbH
+#  All rights reserved.
+#
+#  This program is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License version 2 as published by
+#  the Free Software Foundation.
+#
+#===============================================================================
+
+printf "======================================\n" > /dev/console
+printf "PHYTEC TPM2 Test Provision Version 0.2\n" > /dev/console
+printf "======================================\n" > /dev/console
+
+# Get CPU UID
+get_cpu_uid() {
+    val=$(cat /sys/devices/soc0/soc_uid)
+    echo "${val}"
+}
+
+set -e
+trap end EXIT
+end() {
+    if [ "$?" -ne 0 ]; then
+        printf "PHYTEC-TPM2-Provision:fail! \n" > /dev/console
+    fi
+}
+
+# Variable declaration
+CA_DIR=`mktemp -d -t tpm2tmpca.XXXXXX`
+export CA_DIR
+
+tpm2pkcs11tool='pkcs11-tool --module /usr/lib/pkcs11/libtpm2_pkcs11.so'
+# Generate SO-PIN
+TPM_SOPIN=$(tpm2_getrandom --hex 8 | tr -dc 0-9)
```

Comment:
The SOPIN is not stored anywhere. So it would be lost after the script was executed.

Reply:
yes this is intended, because the SO-PIN is used for token initialization and user pin setting.
So in normal use the SO-PIN is not necessary or a risk. If you create additional slots with token, then you can use a new SO-PIN. Do you have any use case for SO-PIN in normal use (not provisioning)?